src/overlays/actors/ovl_Bg_Hakugin_Post/z_bg_hakugin_post.c

```diff
@@ -21,103 +54,994 @@ const ActorInit Bg_Hakugin_Post_InitVars = {
+    MtxF* matrix = SysMatrix_GetCurrentState();
+
+    matrix->mf[3][0] = arg0->x;
+    matrix->mf[3][1] = arg0->y;
+    matrix->mf[3][2] = arg0->z;
+}
+
+void func_80A9B3BC(BgHakuginPost* this, GlobalContext* globalCtx) {
+    s32 pad;
+    s32 sp28 = BGHAKUGINPOST_GET_7F00(&this->dyna.actor);
+    s32 sp24 = BGHAKUGINPOST_GET_7F00(&this->dyna.actor) + 1;
+    s32 sp20;
+    s32 sp1C;
+
+    if (Flags_GetSwitch(globalCtx, sp28)) {
+        sp20 = true;
+    } else {
+        sp20 = false;
+    }
+
+    if (Flags_GetSwitch(globalCtx, sp24)) {
+        sp1C = true;
+    } else {
+        sp1C = false;
+    }
+
+    if (!(sp20 | sp1C)) {
+        Actor_SetSwitchFlag(globalCtx, sp28);
+        this->unk_170 = true;
+    } else {
+        this->unk_170 = sp20;
+    }
+
+    this->unk_174 = sp1C;
+}
+
+void func_80A9B46C(BgHakuginPost* this, GlobalContext* globalCtx) {
+    s32 pad;
+    s32 sp30 = BGHAKUGINPOST_GET_7F00(&this->dyna.actor);
+    s32 sp2C = BGHAKUGINPOST_GET_7F00(&this->dyna.actor) + 1;
+    s32 sp28;
+    s32 sp24;
+
+    if (Flags_GetSwitch(globalCtx, sp30) != 0) {
+        sp28 = true;
+    } else {
+        sp28 = false;
+    }
+
+    if (Flags_GetSwitch(globalCtx, sp2C) != 0) {
+        sp24 = true;
+    } else {
+        sp24 = false;
+    }
+
+    if ((this->unk_170 == false) && (sp28 == true)) {
+        Actor_UnsetSwitchFlag(globalCtx, sp2C);
+    } else if ((this->unk_174 == false) && (sp24 == true)) {
+        Actor_UnsetSwitchFlag(globalCtx, sp30);
+    }
+
+    this->unk_170 = sp28;
+    this->unk_174 = sp24;
+}
+
+void func_80A9B554(BgHakuginPost* this, GlobalContext* globalCtx, BgHakuginPostUnkStruct* unkStruct,
+                   BgHakuginPostUnkStruct1* unkStruct1) {
+    f32 spE4 = D_80A9D880[unkStruct1->unk_00].unk_00;
+    s32 i;
+    s16 val;
+    BgHakuginPostUnkStruct2* unkStruct2;
+    BgHakuginPostUnkStruct1* unkStruct1Temp;
+    f32 temp_f24;
+    f32 temp_f0;
+    f32 temp_f20;
+    f32 temp_f22;
+    Vec3f spB8;
+    Vec3f spAC;
+    Vec3f spA0;
+    f32 temp_f28;
+    f32 temp_f6;
+    s32 pad[2];
+    s16 temp_s1_3;
+
+    spB8.x = unkStruct1->unk_14.x + this->dyna.actor.world.pos.x;
+    spB8.y = unkStruct1->unk_14.y + this->unk_16C;
+    spB8.z = unkStruct1->unk_14.z + this->dyna.actor.world.pos.z;
+
+    spAC.x = Math_SinS(unkStruct1->unk_28) * unkStruct1->unk_20;
+    spAC.y = unkStruct1->unk_24;
+    spAC.z = Math_CosS(unkStruct1->unk_28) * unkStruct1->unk_20;
+
+    for (i = 0, val = 0; i < 65; i++, val += 0x3F0) {
+        unkStruct2 = &unkStruct->unk_02A4[i];
+        unkStruct2->unk_00 = (Rand_ZeroOne() * 0.038f) + 0.002f;
+        temp_f24 = Math_SinS(val);
+        temp_f28 = Math_CosS(val);
+        temp_f20 = Rand_ZeroOne();
+        temp_f20 = (1.0f - SQ(temp_f20)) * 198.0f;
+        unkStruct2->unk_04.x = (temp_f24 * temp_f20) + spB8.x;
+        unkStruct2->unk_04.y = (Rand_ZeroOne() * spE4) + spB8.y;
+        unkStruct2->unk_04.z = (temp_f28 * temp_f20) + spB8.z;
```

Comment:
```suggestion
        unkStruct2->unk_04.x = temp_f24 * temp_f20 + spB8.x;
        unkStruct2->unk_04.y = Rand_ZeroOne() * spE4 + spB8.y;
        unkStruct2->unk_04.z = temp_f28 * temp_f20 + spB8.z;
```
should still match (applies in other places in this wonderful float function). Maybe change the order of the addition too if possible.